### DISCUSSION TITLE
Getting exceptions instead of friendly warnings

### DISCUSSION THREAD

**Discussion by mfsalomons** [dataplat/dbatools#9383] [Q&A] [answered]
I want exceptions so I can create alert when a script fails. Currently with the code:
```
  $PSDefaultParameterValues = @{ '*-Dba*:EnableException' = $true } # EnableException for all dbatools commands using wildcards
  New-DbaDbUser -SqlInstance $serverconn -Database $databasename -Login test -Username test -EnableException
```
I still get the friendly warning if the user does not exists:
WARNING: [14:51:31][New-DbaDbUser] Invalid Login: [test] is not found on [server], skipping.
A try and catch block ignores this.

How can I catch this error / warning? My attempts to get an exception have failed.

**Comment by mg0x7BE** [comment, marked as answer]
Might be a silly idea, but have you tried setting `$WarningPreference = 'Stop'` (or `'Inquire'`) before your try and catch block?

**Reply by mfsalomons** [reply]
Thank you, I knew it was one small (silly) thing I forgot!
